api/access/server.go

```diff
@@ -128,12 +129,87 @@ func (s *Server) GetLatestBlock(ctx context.Context, in *access.GetLatestBlockRe
 	return nil, errors.New("not implemented")
 }
 
-func (s *Server) GetBlockByID(ctx context.Context, in *access.GetBlockByIDRequest) (*access.BlockResponse, error) {
-	return nil, errors.New("not implemented")
+func (s *Server) GetBlockByID(_ context.Context, in *access.GetBlockByIDRequest) (*access.BlockResponse, error) {
+	blockID := flow.HashToID(in.Id)
+	height, err := s.index.HeightForBlock(blockID)
+	if err != nil {
+		return nil, fmt.Errorf("could not get height for block ID %x: %w", blockID, err)
+	}
+
+	header, err := s.index.Header(height)
+	if err != nil {
+		return nil, fmt.Errorf("could not get header for height %d: %w", height, err)
+	}
+
+	ss, err := s.index.SealsByHeight(height)
+	if err != nil {
+		return nil, fmt.Errorf("could not get seals for height %d: %w", height, err)
+	}
+
+	seals := make([]*entities.BlockSeal, 0, len(ss))
+	for _, sID := range ss {
+		seal, err := s.index.Seal(sID)
+		if err != nil {
+			return nil, fmt.Errorf("could not get seal from ID %x: %w", sID, err)
+		}
+
+		// See https://github.com/onflow/flow-go/blob/v0.17.4/engine/common/rpc/convert/convert.go#L180-L188
+		blockSeal := entities.BlockSeal{
+			BlockId:                    seal.BlockID[:],
+			ExecutionReceiptId:         seal.ResultID[:],
+			ExecutionReceiptSignatures: [][]byte{}, // filling seals signature with zero
+		}
+		seals = append(seals, &blockSeal)
+	}
+
+	cc, err := s.index.CollectionsByHeight(height)
+	if err != nil {
+		return nil, fmt.Errorf("could not get collections for height %d: %w", height, err)
+	}
+
+	collections := make([]*entities.CollectionGuarantee, 0, len(cc))
+	for _, cID := range cc {
+		g, err := s.index.Guarantee(cID)
```

Comment:
You really like taking things to the extreme :D

May I suggest `collID`, `guarantee` and `entity` for this loop body?